client/components/date-range/shortcuts.tsx

```diff
@@ -0,0 +1,97 @@
+import { Button } from '@wordpress/components';
+import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import { useEffect, useState } from 'react';
+
+const DATERANGE_PERIOD = {
+	DAY: 'day',
+	WEEK: 'week',
+	MONTH: 'month',
+};
+
+const DateRangePickerShortcuts = ( {
+	currentShortcut,
+	onClick,
+}: {
+	currentShortcut?: string;
+	onClick: ( shortcut: { newFromDate: moment.Moment; newToDate: moment.Moment } ) => void;
+} ) => {
+	const translate = useTranslate();
+	const [ selectedShortcut, setSelectedShortcut ] = useState( currentShortcut );
+
+	useEffect( () => {
+		setSelectedShortcut( currentShortcut );
+	}, [ currentShortcut ] );
```

Comment:
Can you explain why this effect is necessary? 

On each render the `currentShortcut` is set as the initial state in the `useState`. Then the effect will run and do the exact same thing.

Reply:
I guess it was some interim status tweaking the code. Is it @annacmc 

I remove the code block in 766fe31ba0.